### ISSUE TITLE
AMD MI300X (ROCm 6.4) Training & Evaluation Report — LIBERO Benchmark

### ISSUE THREAD

**Issue by ZJLi2013** [starVLA/starVLA#254]
## Summary

StarVLA works **out-of-box** on AMD Instinct MI300X GPUs with ROCm 6.4. We completed full LIBERO training (20K steps, 8 GPUs, DeepSpeed ZeRO-2) and 3-suite evaluation, achieving results that match the NVIDIA baseline — **with zero source code changes**.

## Setup

| Component | Configuration |
|-----------|--------------|
| GPU | 8× AMD Instinct MI300X (gfx942) |
| Software | ROCm 6.4.3, PyTorch 2.6.0 |
| Docker | `rocm/pytorch:rocm6.4.3_ubuntu22.04_py3.10_pytorch_release_2.6.0` |
| Model | Qwen3-VL-4B + QwenOFT |
| Data | LIBERO 4-in-1 (spatial + object + goal + long) |
| Attention | `sdpa` (PyTorch built-in AOTriton, replaces `flash_attention_2`) |
| Distributed | DeepSpeed ZeRO-2, 8 processes |

The only change needed is a command-line flag:

```bash
--framework.qwenvl.attn_implementation sdpa
```

Everything else (DeepSpeed config, data loading, training loop, eval pipeline) works as-is.

## Training Results

- **20K steps**, 15 hours, loss 0.264 → 0.014 (95% reduction)
- 2 checkpoints saved (10K, 20K)
- 8 GPUs fully utilized (73–100%), no hangs, RCCL communication stable throughout
- AOTriton SDPA backend active for both forward and backward passes

| Step | Loss | Step | Loss |
|------|------|------|------|
| 1K | 0.143 | 10K | **0.044** (ckpt saved) |
| 3K | 0.079 | 14K | 0.030 |
| 5K | 0.068 | 18K | 0.021 |
| 7K | 0.059 | 20K | **0.014** (ckpt saved) |

## Evaluation Results (20K Checkpoint)

| Suite | Success Rate (3 trials/task) | NVIDIA Reference (30K, 50 trials/task) |
|-------|------------------------------|---------------------------------------|
| Spatial | **96.7%** (29/30) | 97.8% |
| Object | **100%** (30/30) | 98.6% |
| Goal | **96.7%** (29/30) | 96.2% |
| **Average** | **97.8%** | **97.5%** |

### Evaluation Videos

**Spatial — pick up bowl from cabinet (success):**

**Object — pick up orange juice (success):**

**Goal — open drawer and put bowl inside (success):**

**Goal — push plate to stove (success):**

### What works out-of-box
- DeepSpeed ZeRO-2 (existing config, no changes)
- RCCL multi-GPU communication (NCCL env vars are compatible)
- Data loading (LeRobot format, torchvision_av backend)
- Model checkpointing and resumption
- Policy server (websocket-based inference)

### Attention backend
PyTorch 2.6+ on ROCm ships with **AOTriton** as the SDPA backend. By setting `attn_implementation: sdpa`, no extra flash-attn package is needed. Alternatively, the AMD flash-attn wheel is available:
```bash
pip install flash-attn --index-url=https://pypi.amd.com/simple --extra-index-url https://pypi.org/simple
```

### LIBERO eval on ROCm (extra setup)
- Use `robosuite==1.4.1` (1.5+ has breaking API changes)
- Set `MUJOCO_GL=osmesa` (EGL not supported in AMD Docker)
- Patch LIBERO's `torch.load()` for PyTorch 2.6 compatibility

### Minor bug found
`starVLA/model/framework/__init__.py` line 33 calls `logger.log()`, but the `PureOverwatch` class only exposes `debug/info/warning/error/critical`. This causes an `AttributeError` when importing framework submodules fails. Fix: `logger.log(...)` → `logger.warning(...)`. (Happy to submit a small fix PR for this.)

### Critical pitfall: `HSA_OVERRIDE_GFX_VERSION`
**Do NOT set `HSA_OVERRIDE_GFX_VERSION=11.0.0`** on MI300X (gfx942). This value targets gfx1100 (RDNA3) and causes immediate GPU hangs with SIGABRT on CDNA3 GPUs.

## Environment

```
PyTorch 2.6.0
ROCm/HIP: 6.4.43482-d68111535
GPU: AMD Instinct MI300X (gfx942), 192GB HBM3
```

Have fun training and deploy starVLA on AMD GPUs

**Comment by JinhuiYE** [comment]
Hi @ZJLi2013, thanks for this incredibly detailed report! Really glad you were able to figure everything out and get starVLA running on AMD MI300X with ROCm 6.4 — the results look great! 🎉

We've added a reference to this issue in the [LIBERO README](https://github.com/starVLA/starVLA/blob/starVLA/examples/LIBERO/README.md) so others can find your guide easily.

Also, thanks for flagging the logger.log() bug — feel free to submit that fix PR!
